errors/src/commonMain/kotlin/dev/icerock/moko/errors/mappers/ExceptionMappersStorage.kt

```diff
@@ -63,45 +66,69 @@ object ExceptionMappersStorage {
+        noinline mapper: (Throwable) -> T
+    ): ExceptionMappersStorage = register(
+        resultClass = T::class,
+        conditionPair = ConditionPair(
+            condition,
+            mapper as ExceptionMapper
         )
+    )
 
-    fun <E : BasicException, T : Any> find(
+    /**
+     * Tries to find mapper for [throwable] instance. First, a mapper with condition is
+     * looked for. If mapper with condition was not found, then a simple mapper is looked for. If
+     * the mapper was not found, it will return null.
+     * If there is no mapper for the [throwable] of class [E] and [E] does't inherits
+     * [kotlin.Exception], then exception will be rethrown.
+     */
+    fun <E : Throwable, T : Any> find(
         resultClass: KClass<T>,
-        exception: E,
+        throwable: E,
         exceptionClass: KClass<out E>
     ): ((E) -> T)? {
-        return conditionMappers.get(resultClass)
-            ?.find { it.condition(exception) }
+        val mapper = conditionMappers.get(resultClass)
+            ?.find { it.condition(throwable) }
             ?.mapper as? ((E) -> T)
             ?: mappersMap.get(resultClass)?.get(exceptionClass) as? ((E) -> T)
+
+        return if (mapper == null && throwable !is Exception) {
+            throw throwable
+        } else {
+            mapper
+        }
     }
 
     /**
-     * Tries to find mapper (E) -> T. First, a mapper with a condition is looked for. If this was
-     * not found, then a simple mapper is looked for.
+     * Tries to find mapper (E) -> T by [throwable] instance. First, a mapper with condition is
+     * looked for. If mapper with condition was not found, then a simple mapper is looked for. If
+     * the mapper was not found, it will return null.
+     * If there is no mapper for the [throwable] of class [E] and [E] does't inherits
+     * [kotlin.Exception], then exception will be rethrown.
      */
-    inline fun <E : BasicException, reified T : Any> find(exception: E): ((E) -> T)? =
-        find(
-            T::class,
-            exception,
-            exception::class
-        )
+    inline fun <E : Throwable, reified T : Any> find(throwable: E): ((E) -> T)? = find(
+        resultClass = T::class,
+        throwable = throwable,
+        exceptionClass = throwable::class
+    )
 
-    fun setUnknownErrorText(text: StringDesc): ExceptionMappersStorage {
-        unknownErrorText = text
-        return this
+    fun <T : Any> setFallbackValue(clazz: KClass<T>, value: T): ExceptionMappersStorage {
+        fallbackValuesMap[clazz] = value
+        return ExceptionMappersStorage
     }
+
+    inline fun <reified T : Any> setFallbackValue(value: T): ExceptionMappersStorage =
+        setFallbackValue(T::class, value)
+
+    fun <T : Any> getFallbackValue(clazz: KClass<T>): T {
+        return fallbackValuesMap[clazz] as? T
+            ?: throw IllegalArgumentException("There is no fallback value for class $clazz")
```

Comment:
i agree what we should throw exception here, but please create our own exception class for this situation. maybe `NoFallbackValueException`?

Reply:
or `FallbackValueNotFoundException`

Reply:
also to prevent crashes in this code at production, i think we should try to get fallback value at first in all mappers like `throwableToStringDesc`. And i think we should change it to `throwableMapper`:
```
object ExceptionMappersStorage {
...
    fun <T : Any> getFallbackValue(): T {
...
    }

    fun <E : Throwable, T : Any> throwableMapper(clazz: KClass<T>): (e: E) -> T {
        val fallback = getFallbackValue<T>()
        return { e ->
            find<E, T>(clazz, e, e::class)?.invoke(e) ?: fallback
        }
    }

    inline fun <E : Throwable, reified T : Any> throwableMapper(): (e: E) -> T {
        return throwableMapper(T::class)
    }
}

fun test1() {
    test2(ExceptionMappersStorage.throwableMapper<Throwable, StringDesc>())
    test3(ExceptionMappersStorage.throwableMapper<Throwable, ImageResource>())
}

fun test2(mapper: (Throwable) -> StringDesc) {

}

fun test3(mapper: (Throwable) -> ImageResource) {

}
```

in this implementation we got crash with `FallbackValueNotFoundException` at startup of application, because getting called at start

Reply:
`FallbackValueNotFoundException` was created. Factory `throwableMapper` also was created.